src/main/scala/mg7/pipeline.scala

```diff
@@ -0,0 +1,209 @@
+package ohnosequences.mg7
+
+import ohnosequences.mg7.loquats._
+import ohnosequences.loquat._
+import ohnosequences.statika._, aws._
+import ohnosequences.datasets._
+import ohnosequences.awstools.s3._
+import com.amazonaws.auth._, profile._
+
+
+trait AnyMG7Pipeline { pipeline =>
+
+  type Parameters <: AnyMG7Parameters
+  val  parameters: Parameters
+
+  val inputSamples: Map[SampleID, S3Resource]
+  val outputS3Folder: (SampleID, StepName) => S3Folder
+
```

Comment:
All the input/output data definitions should be factored into their own classes.